walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -1,23 +1,70 @@
-// Copyright 2019 Google LLC
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     https://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
+
+  /**
+    This function is responsible for adding the "find a meeting" feature.
+    @param events has a collection of the events happening during the day
+    @param request has details of the meeting that is required to be scheduled.
+    @return a Collection that has all the possible timeRanges for te requested meeting.
+  */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    
+    long meetingDuration = request.getDuration();      // This is the duration of the request meeting
+    Collection<String> meetingAttendees = request.getAttendees();    // This is a collection of strings that have the names of all the attendees in the to be set meeting
+
+    if ( (meetingDuration <= TimeRange.WHOLE_DAY.duration()) == false)
+      return Arrays.asList();
+
+    else if(events.isEmpty()) 
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    
+    // This ArrayList contains all the unavailable times of TimeRanges at which the request meeting cannot be set. 
+    // This is calculated or set by taking each event from the collection, and checking if the event attendees are busy in the collection
+    // of meetings, and hence, can't be available for the meeting request.  
+    ArrayList<TimeRange> unavailableTimes = new ArrayList<TimeRange>();
+
+    for (Event event : events){
+      Collection<String> eventAttendees = event.getAttendees();
+      TimeRange rangeOfEvent = event.getWhen();
+     
+      for (String attendee : meetingAttendees) {    // Checking the attendees of the meeting request 
```

Comment:
Comments on the logic of a chunk of code usually goes to its own line at the top as well.